### ISSUE TITLE
Callback is not functioning in windows

### ISSUE THREAD

**Issue by santoshpanda1989** [open62541/open62541#1268]
Description
===========
I want to execute callback mechanism in server_nodeset_plcopen example. I have already done the same in OSX and it works fine. I have to do it in windows but when I run the server in windows machine and try to read the variable node which has been implemented with callback mechanism, there is an access violation error at address 0xCCCCCCCC. I searched the same and found it is an error which happens due to uninitialization of variables. PFA the server_nodeset_plcopen.c file.
[server_nodeset_plcopen_cb.txt](https://github.com/open62541/open62541/files/1397895/server_nodeset_plcopen_cb.txt)

Background Information / Reproduction Steps
===========================================

Checklist
=========
Please provide the following information:

 - [x] open62541 Version (release number or git tag): Master
 - [x] Other OPC UA SDKs used (client or server): UAEXPERT
 - [x] Operating system: Windows 7
 - [ ] Logs (with `UA_LOGLEVEL` set as low as necessary) attached
 - [ ] Wireshark network dump attached
 - [ ] Self-contained code example attached
 - [ ] Critical issue

**Comment by Pro** [comment]
I think your example is not complete, e.g. there is no node with ID `6177` in any of the nodesets.

Addtionally you should always check the return value of the methods, i.e. `UA_Server_setVariableNode_valueCallback` returns in your example the status code `UA_STATUSCODE_BADNODEIDUNKNOWN`.

There is no namespace with index 4 in the example.

Additionally, if you wanted to add a callback to the ServerStatus -> CurrentTime (ns0, i=2258) keep in mind that there is a datasource attached to it (https://github.com/open62541/open62541/blob/8cfce19a030808892947eb86c40d5d75f61cdfb7/src/server/ua_server_ns0.c#L625) so it would be better to test on another variable.

So unfortunately I couldn't reproduce it.

**Comment by santoshpanda1989** [comment]
@Pro I had two additional xml files and I have added them in the cmakelists.txt in the nodeset folder instead of Opc.ua.plcopen.nodeset2.xml. The error was for the uninitialization of "UA_Callback.onwrite". When I provided the "afterWriteTime" function to it then everything worked fine. My question is If I do not have any requirement to write any nodes then do I need to always initialize the "UA_Callback.onwrite".
